taskvine/src/manager/vine_manager.c

```diff
@@ -5239,13 +5233,22 @@ static struct vine_task *vine_wait_internal(struct vine_manager *q, int timeout,
 		int retrieved_this_cycle = 0;
 		BEGIN_ACCUM_TIME(q, time_receive);
 		do {
-			char *key;
-			struct vine_worker_info *w;
+			char *key = NULL;
+			struct vine_worker_info *w = NULL;
 
 			// consider one worker at a time
+			while (list_size(q->workers_with_complete_tasks) > 0) {
+				key = list_pop_head(q->workers_with_complete_tasks);
+				free(key);
+
+				w = hash_table_lookup(q->worker_table, key);
+				if (w) {
```

Comment:
Make this crystal clear that the break is the expected case, and the else is the unexpected case...